micromanager_gui/main_window.py

```diff
@@ -148,92 +159,121 @@ def add_frame_explorer(self, name, array):
         except KeyError:
             self.viewer.add_image(array, name=layer_name)
 
-    # TO DO: add the file name form the save box
-    def _on_mda_frame(self, image, event):
 
-        sequence = event.sequence
+    # create temp folder when mda starts
+    def _on_mda_started_temp_folder(self, sequence: useq.MDASequence):
```

Comment:
```suggestion
    def _on_mda_started(self, sequence: useq.MDASequence):
        """"create temp folder when mda starts."""
```

for a callback like this, name it based on what it responds to, not what it does (you may want to add things other than creating a temp folder later).

also, use docstrings for comments that aren't things like "TODO:"...